### PR TITLE
fix: Handle an electron.d.ts file in a custom build

### PR DESCRIPTION
**Pull request by felixrieseberg**
#### Description of Change
Right now, if you use a [custom build or a mirror during the installation of Electron](https://github.com/electron/get#specifying-a-mirror), you will always get the `electron.d.ts` file that was part of the npm package - and not the one that's part of your custom build or mirror.

With this change, during installation, we check if the downloaded zip file contained an `electron.d.ts` file. If so, we move it one level up. 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed: If a custom Electron build contains an electron.d.ts file, it is now being used.

### CODE REVIEW

**Comment by felixrieseberg** [comment]
The broken Windows build seems to be a fluke!

**Comment by MarshallOfSound** [comment]
@felixrieseberg Can you re-auth with Circle to kick off the CI builds, you may have to push an empty commit too

File: npm/install.js [hunk 1]
```diff
@@ -71,9 +71,26 @@ function isInstalled () {
...
+    const distPath = path.join(__dirname, 'dist')
+
+    extract(zipPath, { dir: distPath }, err => {
       if (err) return reject(err);
 
+      // If the zip contains an "electron.d.ts" file,
+      // move that up
+      const srcTypeDefPath = path.join(distPath, 'electron.d.ts')
+      const targetTypeDefPath = path.join(__dirname, 'electron.d.ts')
+      const hasTypeDefenitions = fs.existsSync(srcTypeDefPath)
```
Comment (codebytere) [inline npm/install.js:83]: nit: `s/hasTypeDefenitions/hasTypeDefinitions`

**Review by codebytere — APPROVED** [review]
Approved w/ small typo fix!

File: npm/install.js (hunk 1 above)
Comment (felixrieseberg) [inline npm/install.js:83]:
```suggestion
      const hasTypeDefinitions = fs.existsSync(srcTypeDefPath)
```

**Comment by jkleinsc** [comment]
@felixrieseberg can you re-authenticate with CircleCI and then push an empty commit to trigger CircleCI testing?

**Review by jkleinsc — CHANGES_REQUESTED** [review]
Lint failures need to be addressed.

**Comment by ckerr** [comment]
@jkleinsc are you good with these lint changes?

**Review by jkleinsc — APPROVED** [review]

**Comment by hassanmahdi58** [comment]
electron.d.ts

**Pull request merged**